src/TVShowTime/Provider.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace SocialiteProviders\TVShowTime;
+
+use SocialiteProviders\Manager\OAuth2\User;
+use Laravel\Socialite\Two\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'TVSHOWTIME';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['basic'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://www.tvshowtime.com/oauth/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://api.tvshowtime.com/v1/oauth/access_token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get('https://api.tvshowtime.com/v1/user', [
+            'headers' => [
+                'TVST_ACCESS_TOKEN' => $token,
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['user']['id'],
+            'nickname' => $user['user']['name'],
+            'avatar'   => $user['user']['image'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```

Comment:
@Henokv Could you please remove this method as you are not using it for anything.